src/cpp/src/continuous_batching_impl.cpp

```diff
@@ -148,8 +148,10 @@ void ContinuousBatchingPipeline::ContinuousBatchingImpl::step() {
     if (scheduler_output.m_total_num_scheduled_tokens == 0) {
         for (size_t i = 0; i < m_requests.size(); ++i) {
             SequenceGroup::Ptr sequence_group = m_requests[i];
-            sequence_group->set_out_of_memory();
-            sequence_group->notify_handle();
+            if (!sequence_group->is_waiting()) {
+                sequence_group->set_out_of_memory();
+                sequence_group->notify_handle();
+            }
```

Comment:
in that change https://github.com/openvinotoolkit/openvino.genai/pull/907/files#diff-477d5a9eeff46e99fa6e2bf34e89d113b6533b0056b2db3670e18dd89f25a4aaR548 you have changed `is_waiting` to return `m_is_gen_paused` instead of `false`

Why was it required? Without that change we don't need this update, because `is_waiting` status is cleaned within `schedule` here  
https://github.com/openvinotoolkit/openvino.genai/blob/275729c96a66e9eac8d09792fe62c1314a2bb69b/src/cpp/src/scheduler.hpp#L67

Reply:
maybe we can rename `is_waiting` to `is_preempted` for better clarify?

Reply:
1. Yes, I am agree to rename `is_waiting ` to `is_preempted`. It will be more clear. 
2. Related to "return m_is_gen_paused instead of false" -> it was implemented not to schedule tokens in case of request was paused

Reply:
ok, I see